src/utils/ChildMapping.js

```diff
@@ -0,0 +1,87 @@
+import { Children } from 'react';
+
+
+/**
+ * Given `this.props.children`, return an object mapping key to child.
+ *
+ * @param {*} children `this.props.children`
+ * @return {object} Mapping of key to child
+ */
+export function getChildMapping(children) {
+  if (!children) {
+    return children;
+  }
+  let result = {};
+  Children.map(children, child => child)
+    .forEach((child) => {
+      result[child.key] = child;
+    });
```

Comment:
I don't think so, or at least not fully. 

from what I gather, `flattenChildren` will return  object where the keys are the react key/data-id and values the corresponding children. my code above is attempting to emulate it by running it through `Children.map` which flattens and  adds keys to unkeyed elements. The forEach turns it into a hash.

The double iteration isn't great, but also we lose the extra debug info that `flattenChildren` gave, pointing out duplicates keys.